src/lib/storage/storage_manager.cpp

```diff
@@ -30,123 +34,149 @@ void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> t
   }
 
   // Create table statistics and chunk pruning statistics for added table.
+
   table->set_table_statistics(TableStatistics::from_table(*table));
   generate_chunk_pruning_statistics(table);
 
-  _tables.emplace(name, std::move(table));
+  if (table_iter == _tables.end())
+    _tables.emplace(name, std::move(table));
+  else
+    _tables[name] = std::move(table);
```

Comment:
Wouldn't this line suffice? If not, please add curly braces

Reply:
Good point, I don't know why I wanted to reuse the old solution.